facebook/qsf-tools/generate-codebook.R

```diff
@@ -0,0 +1,360 @@
+#!/usr/bin/env Rscript
+
+## Add information from one .qsf translation file at a time to the survey codebook
+##
+## Usage:
+##
+## Rscript generate-codebook.R path/to/qsf path/to/codebook
+
+suppressPackageStartupMessages({
+  library(tidyverse)
+  library(jsonlite)
+  library(stringr)
+  library(gsubfn)
+  source("qsf-utils.R")
+})
+
+
+process_qsf <- function(path_to_qsf,
+                        path_to_shortname_map="./static/item_shortquestion_map.csv",
+                        path_to_replacement_map="./static/item_replacement_map.csv") {
+  q <- read_json(path_to_qsf)
+  
+  # get the survey elements that are questions:
+  ii_questions <- q$SurveyElements %>% 
+    map_chr("Element") %>%
+    {. == "SQ"} %>% 
+    which()
+  
+  # get the questions that were shown to respondents
+  shown_items <- get_shown_items(q)
+  ii_shown <- q$SurveyElements[ii_questions] %>% 
+    map_chr(~ .x$Payload$QuestionID) %>%
+    {. %in% shown_items} %>% 
+    which()
+  
+  # subset qsf to valid elements
+  displayed_questions <- q$SurveyElements[ii_questions][ii_shown]
+  
+  # Qualtrics auto-assigned question IDs
+  qids <- displayed_questions %>% 
+    map_chr(~ .x$Payload$QuestionID)
+  
+  # the items are how we will match these to the survey data:
+  items <- displayed_questions %>% 
+    map_chr(~ .x$Payload$DataExportTag)
+  
+  # B13 was originally named incorrectly. Rename manually as needed
+  items[items == "B13 "] <- "B13"
+  
+  # get the text of the question:
+  questions <- displayed_questions %>% 
+    map_chr(~ .x$Payload$QuestionText) %>% 
+    str_remove_all("<[^<>]+>") %>% 
```

Comment:
I realize that the survey only contains limited HTML and this is completely harmless here, but I can't resist linking this https://stackoverflow.com/a/1732454

Reply:
lol a classic :joy: